### DISCUSSION TITLE
Sub-Buttons with templating

### DISCUSSION THREAD

**Discussion by svb91** [Clooos/Bubble-Card#484] [Feature requests]
It would be great if I could add a templated value to a sub-button like this:

- show_name: true
  show_last_updated: true
  name: >
    [[[ return states['sensor.og_as_h_heizung_temperature'].state + 'kW']]]
  icon: mdi:home-thermometer-outline

The reason for that is, that my temperature entity does provide this information with 2 digits and I'd like to round the value before displaying it.

**Comment by svb91** [comment]
In general would be templating great (also in the style tag) like this:

.bubble-sub-button-1
{
  {% if 1 < 2 %}
    display: none !important;
  {% endif %}
}

**Comment by lukas-kraus** [comment]
Would love that as well

**Comment by Clooos** [comment]
Hi, go take a look at the Templates section in the documentation, Bubble Card doesn't supports Jinja templates but you can add JS templates, I've added many examples.

**Comment by x2on** [comment]
Any idea to achieve rounding values in sub buttons, without creating rounded values?

**Comment by michaelblight** [comment]
It's just Javascript so you can use `Math.round`, for example:
```
.bubble-sub-button-1 {
  display: ${ Math.round(2.1) == 2 ? "none!important" : "" }
}
```
If the value is coming from a state you'd need to use `parseFloat()` to convert it to a number first.

**Comment by Clooos** [comment]
Hi! Jinja templates will finally be available in v3.4.0! You will be able to use these to customize names, icons, sub-buttons and styles 🍻

**Reply by Clooos** [reply]
And done! It was about time! 🍻

https://github.com/Clooos/Bubble-Card/releases/tag/v3.4.0
